co-authors-plus.php

```diff
@@ -713,12 +717,12 @@ function posts_where_filter( $where, $query ) {
 				}
 				$terms_implode = rtrim( $terms_implode, ' OR' );
 
-				$id = is_author() ? get_queried_object_id() : '\d+';
+				$id = is_author() && $query->is_main_query() ? get_queried_object_id() : '\d+';
```

Comment:
This fix works without the check for `$query->is_main_query()` so I think we don't need to add it for now.

Reply:
If you query a second user from author.php to show something like, people in same dept or other authors you may like, you won't get the correct posts, you'll only get those where the user is in the post_author column.

e.g. If I add `$x = new WP_Query( array ( 'post_status' => 'publish', 'author' => 9099999999 ) );` to author.php after the main loop. Go to `/?author=1` the ID that's passed to the regex is **1** not **9099999999** so our custom query is missing any posts where the queried user isn't in the `post_author` column.

Reply:
Thinking this through you also need this check to avoid further risk of WordPress throwing db errors. If your custom queried user ID contains the author page user ID somewhere before the 3rd char from the end then we'd end up with a split user ID and invalid SQL. 

e.g Go to /?author=1 and sub-query for author 10123456. That'd produce regex that's looking through the the SQL for `\s*(.*1.)/` which would match 1012 from our ID and put the new SQL in the middle of the ID and thus a db error would be thrown.